lib/routes/cspengyuan/index.ts

```diff
@@ -0,0 +1,309 @@
+
+    const items = await Promise.all(
+        (list as any[])
+            .filter((l) => l.link !== null)
+            .map((item) =>
+                cache.tryGet(item.link, async () => {
+                    if (category === 'publication') {
+                        const response = await got(item.link, { agent: { https: agent } });
+                        const content = load(response.body);
+                        const p = content('div.mrh-dtl-right-top > p');
+                        const b = content('div.mrh-dtl-right-bom');
+                        const imgUrl = content('img').attr('src');
+                        if (type === 'periodical') {
+                            item.description = `
+                        <h4>${content(p[0]).text().trim()}</h4>
+                        <h4>${content(p[1]).text().trim()}</h4>
+                        <a download="${item.pdfName}" href="https://www.cspengyuan.com${item.pdfUrl}"
+                        style="display: inline-block; padding: 10px 20px; background-color: red; color: white;
+                        text-align: center; text-decoration: none; border-radius: 5px; font-size: 16px;">整刊下载</a><br>
+                        <img src="${imgUrl}" height="50%" style="display: block; margin: 0 auto;">
+                    `;
+                        } else if (type === 'monograph') {
+                            item.description = `
+                        <h4>${content(p[0]).text().trim()}</h4>
+                        <h4>${content(p[1]).text().trim()}</h4>
+                        <h4><b>${b.find('h4 > b').text().trim()}</b></h4>
+                        <p>${b.find('p').text().trim()}</p><br>
+                        <img src="${imgUrl}" height="50%" style="display: block; margin: 0 auto;">
+                    `;
+                        }
+                    } else {
+                        const image = ctx.req.query('image') ?? false;
+                        let screenshotBase64;
+                        if (image) {
+                            const browser = await puppeteer();
+                            const page = await browser.newPage();
+                            await page.goto(item.pdfViewUrl, { waitUntil: 'networkidle0' });
+                            logger.http(`Requesting ${item.pdfViewUrl}`);
+                            screenshotBase64 = await page.screenshot({
+                                encoding: 'base64',
+                            });
+                            await browser.close();
+                        }
+                        item.description = `
+                            pdf原链接: <a download="${item.pdfName}" href="https://www.cspengyuan.com${item.pdfUrl}">Download</a><br>
+                            pdf在线预览: <a href="${item.pdfViewUrl}">预览</a><br>
+                            <img src="data:image/png;base64,${screenshotBase64}" style="${screenshotBase64 ? '' : 'display: none'}">
+                            `;
```

Comment:
Do not provide the PDF screenshot as it makes RSSHub response huge. Providing a link to the PDF is sufficient.